### ISSUE TITLE
ALPSCore_DIR environment variable issues

### ISSUE THREAD

**Issue by ryanlevy** [ALPSCore/ALPSCore#427]
As started in PR #425, `ALPSCore_DIR=$root` doesn't seem to do the trick for allowing it to install.
Setup:
```bash
$ CC=clang CXX=clang++ cmake .. -DENABLE_MPI=$ENABLE_MPI -DCMAKE_INSTALL_PREFIX=~/modules/ALPSCore/clang/build -DCMAKE_BUILD_TYPE=Release -DEIGEN3_INCLUDE_DIR=$EIGEN3_INCLUDE_DIR -DDocumentation=OFF
$ make
$ make test
$ make install
```

`~/modules/ALPSCore/clang/build/share/ALPSCore` is where `ALPSCoreConfig.cmake` lives

```bash
$ echo $ALPSCore_DIR

$ export ALPSCore_DIR=~/modules/ALPSCore/clang/build
$ cmake .
...
CMake Error at /Users/[me]/modules/ALPSCore/clang/build/ALPSCoreConfig.cmake:60 (find_package):
  Could not find a package configuration file provided by "alps-params" with
  any of the following names:

    alps-paramsConfig.cmake
    alps-params-config.cmake

  Add the installation prefix of "alps-params" to CMAKE_PREFIX_PATH or set
  "alps-params_DIR" to a directory containing one of the above files.  If
  "alps-params" provides a separate development package or SDK, be sure it
  has been installed.
Call Stack (most recent call first):
  /Users/[me]/modules/ALPSCore/clang/build/ALPSCoreConfig.cmake:113 (alpscore_config_main_)
  CMakeLists.txt:14 (find_package)
```

Files:
[CMakeCache_ALPSCore.txt](https://github.com/ALPSCore/ALPSCore/files/1632819/CMakeCache_ALPSCore.txt)
[CMakeCache_ini_params.txt](https://github.com/ALPSCore/ALPSCore/files/1632822/CMakeCache_ini_params.txt)

**Comment by galexv** [comment]
This may happen if the installation directory is the same as the build directory. I am looking at it now...

**Comment by ryanlevy** [comment]
For me, the above still remains true if I move to a build directory and run `cmake ..` from there

**Comment by egull** [comment]
I can confirm that – I have the same problem...

**Comment by galexv** [comment]
I can consistently reproduce it if my `CMAKE_INSTALL_PREFIX` is the same as the ALPSCore build directory. Therefore, a workaround for now is to install into some other directory (e.g., a subdirectory of `build/`), such as:
```
$ cmake ..  -DCMAKE_INSTALL_PREFIX=$PWD/install -DEIGEN3_INCLUDE_DIR=$EIGEN3_INCLUDE_DIR
$ make
$ make install
$ export ALPSCore_DIR=$PWD/install

$ cd /my_app/build
$ cmake .. 
```

**Comment by galexv** [comment]
The problem is that there is a "stray" `ALPSCoreConfig.cmake` in the build directory (which is copied to its proper location by `make install`) which contains paths that are relative to the its final location, not to the build directory. `find_package(ALPSCore ...)` picks up this stray file, and naturally cannot find any components in the expected places.
Again, workaround, till i fix this: do not install into the build directory.

**Comment by egull** [comment]
I don't think you're on the right path. I'll show you an installation tomorrow that has this problem.

**Comment by galexv** [comment]
> For me, the above still remains true if I move to a build directory and run cmake .. from there

Just to clarify: it's not about the location of the build directory, it's about the location of the `${CMAKE_INSTALL_PREFIX}` directory, or, more precisely, about the environment variable `ALPSCore_DIR` that, during the build of a dependent project, is pointing to a directory that contains out-of-place `ALPSCoreConfig.cmake` (such as ALPSCore build directory). 

> I don't think you're on the right path. I'll show you an installation tomorrow that has this problem.

There may be other problems too, but if i move the out-of-place `ALPSCoreConfig.cmake` away from ALPSCore `build` and clear the `CMakeCache.txt` of the dependent project, the dependent project does build in @ryanlevy's configuration. So, avoiding generating the out-of-place `ALPSCoreConfig.cmake` (or making it work regardless of its location) should fix at least one problem. :)

**Comment by ryanlevy** [comment]
Once I deleted the `ALPSCoreConfig.cmake` from the build directory, the PR seems to have worked! On the other hand, I'm fine if the correct set of steps is having both an install and a build directory, it just isn't explicitly in the install wiki so I never noticed.

Also for future bug readers, in the [Jenkins build script](https://github.com/ALPSCore/ALPSCore/blob/master/common/build/build.jenkins.sh) it clearly splits build and install

**Comment by egull** [comment]
Indeed, this looks like two different problems. I propose we merge this PR because it fixes Ryan's problems, and I'll open a new bug with my problem which seems to be different.

**Comment by galexv** [comment]
Closed by PR #455
